src/epochs.lua

```diff
@@ -116,7 +116,8 @@ end
 --- @param epochIndex number The epoch index
 --- @return table<WalletAddress, WalletAddress> # The prescribed observers for the epoch
 function epochs.getPrescribedObserversForEpoch(epochIndex)
-	return epochs.getEpoch(epochIndex).prescribedObservers or {}
+	local epoch = epochs.getEpoch(epochIndex)
+	return epoch and epoch.prescribedObservers or {}
```

Comment:
I'm assuming that returning empty table rather than nil for this and all the below is preferable for flow control and logic management?

Reply:
correct - right now, it simplifies logic when traversing these tables. the impact of changing them to `nil` would be larger change, but not unreasonable.